detox/test/e2e/04.assertions.test.js

```diff
@@ -1,91 +1,56 @@
     await element(by.text('Assertions')).tap();
+  },
+  get textElement() { return element(by.id('main-text')) },
+  get invisibleTextElement() { return element(by.id('offscreen-text')) },
+  get toggleElement() { return element(by.id('toggle')) },
+}
+
+describe('Assertions', () => {
+  beforeEach(async () => {
+    await driver.navToScreen();
   });
 
   it('should assert an element is visible', async () => {
-    await expect(element(by.id('UniqueId204'))).toBeVisible();
+    await expect(driver.textElement).toBeVisible();
   });
 
   it('should assert an element is not visible', async () => {
-    await expect(element(by.id('UniqueId205'))).not.toBeVisible();
+    await expect(driver.invisibleTextElement).not.toBeVisible();
   });
 
-  // prefer toBeVisible to make sure the user actually sees this element
   it('should assert an element exists', async () => {
-    await expect(element(by.id('UniqueId205'))).toExist();
+    await expect(driver.invisibleTextElement).toExist();
   });
 
   it('should assert an element does not exist', async () => {
     await expect(element(by.id('RandomJunk959'))).not.toExist();
   });
 
-  // matches specific text elements like UIButton, UILabel, UITextField or UITextView, RCTText
   it('should assert an element has text', async () => {
-    await expect(element(by.id('UniqueId204'))).toHaveText('I contain some text');
+    await expect(driver.textElement).toHaveText('I contain some text');
   });
 
-  // matches by accessibility label, this might not be the specific displayed text but is much more generic
   it('should assert an element has (accessibility) label', async () => {
-    await expect(element(by.id('UniqueId204'))).toHaveLabel('I contain some text');
+    await expect(driver.textElement).toHaveLabel('I contain some text');
   });
 
   it('should assert an element has (accessibility) id', async () => {
-    await expect(element(by.text('I contain some text'))).toHaveId('UniqueId204');
+    await expect(element(by.text('I contain some text'))).toHaveId('main-text');
   });
 
-  // for example, the value of a UISwitch in the "on" state is "1"
   it.skip(':ios: should assert an element has (accessibility) value', async () => {
-    await expect(element(by.id('UniqueId146'))).toHaveValue('0');
-    await element(by.id('UniqueId146')).tap();
-    await expect(element(by.id('UniqueId146'))).toHaveValue('1');
+    await expect(driver.toggleElement).toHaveValue('0');
+    await driver.toggleElement.tap();
+    await expect(driver.toggleElement).toHaveValue('1');
   });
 
   it('assert toggle-switch widget', async () => {
-    await expect(element(by.id('UniqueId146'))).toHaveToggleValue(false);
-    await element(by.id('UniqueId146')).tap();
-    await expect(element(by.id('UniqueId146'))).toHaveToggleValue(true);
-    await expect(element(by.id('UniqueId146'))).not.toHaveToggleValue(false);
-  });
-
-  it('should throw exception for visibility threshold out of range (lower than 1)', async () => {
```

Comment:
You removed this because those tests didn't actually work, right?

Reply:
Because the use cases are fully covered under `expect` tests and there's no added value in running them as E2E's using futile matchers